### PR TITLE
[5.4] Fix Apple Silicon build

### PR DESCRIPTION
**Pull request by shahmishal**
rdar://75043262

### CODE REVIEW

**Comment by shahmishal** [comment]
@swift-ci test Apple Silicon

**Comment by shahmishal** [comment]
@swift-ci test

**Comment by shahmishal** [comment]
This pull request is focus on fixing the build failure. However, after fixing the build failure now we are seeing following test failures. 
```01:56:00 Failed Tests (18):
01:56:00   Swift(macosx-arm64) :: execution/arc_36509461.swift
01:56:00   Swift(macosx-arm64) :: compiler_scale/scale_neighbouring_getset.gyb
01:56:00   Swift(macosx-arm64) :: stdlib/Reflection_objc.swift
01:56:00   Swift(macosx-arm64) :: Sanitizers/tsan.swift
01:56:00   Swift(macosx-arm64) :: Sanitizers/tsan-norace-deinit-run-time.swift
01:56:00   Swift(macosx-arm64) :: Sanitizers/tsan-norace-block-release.swift
01:56:00   Swift(macosx-arm64) :: Sanitizers/tsan-libdispatch.swift
01:56:00   Swift(macosx-arm64) :: Sanitizers/tsan-emptyarraystorage.swift
01:56:00   Swift(macosx-arm64) :: Sanitizers/symbolication.swift
01:56:00   Swift(macosx-arm64) :: Sanitizers/asan_recover.swift
01:56:00   Swift(macosx-arm64) :: Sanitizers/asan.swift
01:56:00   Swift(macosx-arm64) :: Sanitizers/asan-globals-overwrite.swift
01:56:00   Swift(macosx-arm64) :: Interpreter/indirect_enum.swift
01:56:00   Swift(macosx-arm64) :: IRGen/opaque_result_type_availability.swift
01:56:00   Swift(macosx-arm64) :: IRGen/macosx-sdk-version.swift
01:56:00   Swift(macosx-arm64) :: IRGen/fixed_class_initialization.swift
01:56:00   Swift(macosx-arm64) :: ClangImporter/availability_macosx_canonical_versions_appext.swift
01:56:00   Swift(macosx-arm64) :: ClangImporter/availability_macosx_canonical_versions.swift
```

File: benchmark/README.md [hunk 1]
```diff
@@ -84,7 +85,7 @@ Build steps (with example options):
 
 1. `$ mkdir build; cd build`
 2. `$ cmake [path to swift src]/benchmark -G Ninja -DSWIFT_EXEC=[path to built swiftc]`
-3. `$ ninja swift-benchmark-macosx-x86_64`
+3. `$ ninja swift-benchmark-macosx-$(uname -m)`
```
Comment (porglezomp) [inline benchmark/README.md:88]: I was going to object that `uname` isn't on Windows, but looking at the docs they all seem to just be optimized for mac development anyway, so this is fine.

**Review by porglezomp — APPROVED** [review]

**Review by edymtt — APPROVED** [review]

**Review by tkremenek — APPROVED** [review]

**Pull request merged**